sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotUseRandom.cs

```diff
@@ -30,33 +30,53 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class DoNotUseRandom : SonarDiagnosticAnalyzer
+    public sealed class DoNotUseRandom : HotspotDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S2245";
         private const string MessageFormat = "Make sure that using this pseudorandom number generator is safe here.";
 
         private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager)
+                .WithNotConfigurable();
+
+        public DoNotUseRandom()
+            : base(new DefaultAnalyzerConfiguration())
+        {
+        }
+
+        public DoNotUseRandom(IAnalyzerConfiguration analysisConfiguration)
+            : base(analysisConfiguration)
+        {
+        }
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
         protected override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            context.RegisterCompilationStartAction(
+                ccc =>
```

Comment:
why `ccc`? would be more readable to have a name

Reply:
That sounds like a good reason for not using so many nested lambda expressions!

Reply:
this sounds like a new rule to me